### DISCUSSION TITLE
Updating Custom Attribute IDs in Links During Copy Operation

### DISCUSSION THREAD

**Discussion by goyaljai1** [clientIO/joint#2791] [Q&A] [answered]
### Introduction

I've added a custom attribute to a `Link` element in my project. This attribute stores the IDs of certain elements that the link is associated with (these are separate from the source and target elements directly connected by the link).

When I use `Ctrl + C` to copy a link, I want the copied link’s custom attribute to automatically update with the new IDs of the associated elements that are also being copied to the clipboard.

To clarify:

- The link has a custom attribute that stores IDs of other elements (not the source and target).
- When I copy the link, these associated elements are also copied.
- **I need the new, copied link to reference the IDs of the newly created elements instead of the original IDs.**

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
I assume you use the `ui.Clipboard` plugin.

Can you just update the ids when you paste them? For instance:

```ts
// first paste the clipboard to a tmp graph
const cells = clipboard.pasteElements(new dia.Graph());
updateCustomIds(cells);
// now add modified cells into actual graph
graph.addCells(cells);
```

Note: If you need a map of the original cells to the cloned elements,  you could use `graph.cloneSubgraph()` method.

**Reply by goyaljai1** [reply]
Thanks for the response. Is there a way to map the original cells in a graph to their corresponding copied elements after using `clipboard.copyElements()`?

**Reply by kumilingus** [reply]
No, it's currently not possible. The upcoming version will have a new `cloneCells()` option that allows the user to control how cells are cloned.
